plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchTaskHandler.groovy

```diff
@@ -115,13 +116,14 @@ class AzBatchTaskHandler extends TaskHandler implements FusionAwareTask {
         final done = taskState0(taskKey)==BatchTaskState.COMPLETED
         if( done ) {
             // finalize the task
-            task.exitStatus = readExitFile()
+            final info = batchService.getTask(taskKey).executionInfo
+            task.exitStatus = info?.exitCode ?: readExitFile()
             task.stdout = outputFile
             task.stderr = errorFile
             status = TaskStatus.COMPLETED
-            final info = batchService.getTask(taskKey).executionInfo
-            if (info.result == BatchTaskExecutionResult.FAILURE)
+            if (info.result == BatchTaskExecutionResult.FAILURE && task.exitStatus == 0) {
```

Comment:
This is really counter intuitive, let's add a comment why throw the exception only when exitStatus == 0

Reply:
Added in reverted commit